From 27fe0c490aae752d8cd70b2a5e377cbe75b6e4f0 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 13 Jan 2021 01:46:21 +0200
Subject: [PATCH] [SW-0] preboot: fix high limits in overlap check

The high limits should be included in the overlap check. Otherwise we
get a poorer precision of the overlapped addresses.
Specifically, the stack pointer, points to the most recently pushed
object on the stack and therefor the __stack_top is an entry before
anything was pushed to the stack and can be included.
The "addr + len" expression can be the upper limit because the resulting
offset is never accessed.
This precision will be needed later for addition of the SRAM zeroing
code.

Change-Id: Iadafe106c84094d55b1d08178c2e78b4f2a3cdd8
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/preboot.c b/preboot/preboot.c
index 568785c..3858e65 100644
--- a/preboot/preboot.c
+++ b/preboot/preboot.c
@@ -220,7 +220,7 @@
 
 bool preboot_sram_is_valid_address_range(u64 addr, int len)
 {
-	if (addr >= SDRAM_BASE && addr + len < SDRAM_BASE + SRAM_SIZE)
+	if (addr >= SDRAM_BASE && addr + len <= SDRAM_BASE + SRAM_SIZE)
 		return !preboot_is_overlap_address_range(addr, len);
 
 	return false;
